lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,40 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          tokens = {}
+          Array(config).each do |secret|
+            key, secret = secret.split('=').map(&:strip)
+            pull_request = self.data.pull_request ? self.data.pull_request : ""
+            now = Time.now.to_i()
+            payload = {
+              "iss" => "Travis CI, GmbH",
+              "slug" => self.data.slug,
+              "pull-request" => pull_request,
+              "exp" => now+5400,
+              "iat" => now
+            }
+            begin
+              tokens[key] = JWT.encode(payload, secret)
+            rescue Exception => e
+              sh.failure "JWT Encode Error: #{e.message}"
```

Comment:
Note that `sh.failure` would halt the build immediately. See https://staging.travis-ci.org/travis-ci/travis_staging_test/builds/507053#L138.

I tend to think this is not the desired behavior. I think is better to just inform the user of the error with `sh.echo` (the usefulness of the message `e.message` is dubious, as you can see from the stack trace above; `no implicit conversion of nil into String` does not tell the user what to do) and move on.

A useful message might be something like:

    There was an error while encoding JWT. If the secret is encrypted, ensure that it is encrypted correctly.